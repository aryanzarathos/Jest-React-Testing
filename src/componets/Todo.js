import React from 'react'

const Todo = ({data}) => {
    return (
        <React.Fragment>
            {
                data.map((item, key) => {
                    return (
                        <td key={key}>
                            <tr data-testid={"todo-1"}>{item.id}</tr>
                            <tr data-testid={"todo-2"}>{item.name}</tr>
                            <tr data-testid={"todo-3"}>{item.completed}</tr>
                        </td>
                    )
                })
            }
        </React.Fragment>
    )
}

export default Todo