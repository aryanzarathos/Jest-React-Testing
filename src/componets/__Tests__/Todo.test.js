import { render ,screen , cleanup} from "@testing-library/react";
import Todo from "../todo";
import renderer from "react-test-renderer"

let data=[
    {
      name:"first Todo",
      id:1,
      completed:false
      
    }
  ]
  afterEach(()=>{
    cleanup()
  })
test('should_render_todo-1', () => {
    render(<Todo data={data} />);
    const todoelement=screen.getByTestId('todo-1');
    expect(todoelement).toBeInTheDocument();
    expect(todoelement).toHaveTextContent(1)
})
test('should_render_todo-2', () => {
    render(<Todo data={data} />);
    const todoelement=screen.getByTestId('todo-2');
    expect(todoelement).toBeInTheDocument();
    expect(todoelement).toHaveTextContent("first Todo")
})
test('should_render_todo-3', () => {
    render(<Todo data={data} />);
    const todoelement=screen.getByTestId('todo-3');
    expect(todoelement).toBeInTheDocument();
    // expect(todoelement).toHave(false)
})
test('matches-snapshot', () => {
    const tree=renderer.create(<Todo data={data}/>).toJSON()
    console.log(tree)
    expect(tree).toMatchSnapshot()
})