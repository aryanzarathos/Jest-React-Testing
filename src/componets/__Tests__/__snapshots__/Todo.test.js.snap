// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches-snapshot 1`] = `
<td>
  <tr
    data-testid="todo-1"
  >
    1
  </tr>
  <tr
    data-testid="todo-2"
  >
    first Todo
  </tr>
  <tr
    data-testid="todo-3"
  />
</td>
`;
